main.tf

```diff
@@ -44,6 +44,49 @@ resource "aws_lambda_function" "vpc" {
   tags = "${merge(var.tags, map("Name", "${var.name_prefix}"))}"
 }
 
+resource "aws_lambda_function" "main_s3" {
+  count         = "${var.attach_vpc_config == "false" && var.s3_config == "true" ? 1 : 0}"
```

Comment:
Have you tested that `s3_key` and `s3_bucket` does not interpret `""` as a nil value? Asking because, if it does, we don't need the if-else statements here and the two new resources here.

Reply:
Don't think we can get rid of the two new resources because of Terraform error with:
`"filename": conflicts with s3_bucket` 
`"s3_bucket": conflicts with filename`

Reply:
Thanks for checking @mhd999 - then we'll need this until `0.12` comes out, which hopefully fixes these kinds of issues 👍